models/logs.js

```diff
@@ -155,9 +157,95 @@ const fetchLastAddedCacheLog = async (id) => {
   }
 };
 
+const fetchAllLogs = async (query) => {
+  try {
+    let { type, prev, next, page, size = SIZE, format } = query;
+    size = parseInt(size);
+    page = parseInt(page);
+
+    try {
+      let requestQuery = logsModel;
+
+      if (type) {
+        const logType = type.split(",");
+        if (logType.length >= 1) requestQuery = requestQuery.where("type", "in", logType);
+      }
+
+      requestQuery = requestQuery.orderBy("timestamp", "desc");
+      let requestQueryDoc = requestQuery;
+
+      if (prev) {
+        requestQueryDoc = requestQueryDoc.limitToLast(size);
+      } else {
+        requestQueryDoc = requestQueryDoc.limit(size);
+      }
+
+      if (page) {
+        const startAfter = (page - 1) * size;
+        requestQueryDoc = requestQueryDoc.offset(startAfter);
+      } else if (next) {
+        const doc = await logsModel.doc(next).get();
+        requestQueryDoc = requestQueryDoc.startAt(doc);
+      } else if (prev) {
+        const doc = await logsModel.doc(prev).get();
+        requestQueryDoc = requestQueryDoc.endAt(doc);
+      }
```

Comment:
NIT: can this two if conditions block merged into one?

Reply:
both of them have different logics to be implemented, that's why kept it separated?